app/services/GuaranteeBalanceService.scala

```diff
@@ -18,20 +18,82 @@ package services
 
 import akka.actor.ActorSystem
 import akka.pattern.after
+import config.FrontendAppConfig
 import connectors.GuaranteeBalanceConnector
+import controllers.routes
+import handlers.GuaranteeBalanceResponseHandler
+import javax.inject.Inject
 import models.backend.{BalanceRequestPending, BalanceRequestResponse}
-import models.values.BalanceId
+import models.requests.{BalanceRequest, DataRequest}
+import models.values._
+import org.joda.time.LocalDateTime
+import pages.{AccessCodePage, EoriNumberPage, GuaranteeReferenceNumberPage}
+import play.api.Logging
+import play.api.http.Status.TOO_MANY_REQUESTS
+import play.api.mvc.Result
+import play.api.mvc.Results.Redirect
 import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
-import javax.inject.Inject
-import models.requests.BalanceRequest
+import uk.gov.hmrc.mongo.lock.MongoLockRepository
+import viewModels.audit.AuditConstants.{AUDIT_DEST_RATE_LIMITED, AUDIT_ERROR_RATE_LIMIT_EXCEEDED, AUDIT_TYPE_GUARANTEE_BALANCE_RATE_LIMIT}
+import viewModels.audit.{ErrorMessage, UnsuccessfulBalanceAuditModel}
 
 import scala.concurrent.duration._
 import scala.concurrent.{ExecutionContext, Future}
 
-class GuaranteeBalanceService @Inject() (val actorSystem: ActorSystem, val connector: GuaranteeBalanceConnector)(implicit ec: ExecutionContext) {
+class GuaranteeBalanceService @Inject() (actorSystem: ActorSystem,
+                                         connector: GuaranteeBalanceConnector,
+                                         responseHandler: GuaranteeBalanceResponseHandler,
+                                         auditService: AuditService,
+                                         mongoLockRepository: MongoLockRepository,
+                                         config: FrontendAppConfig
+)(implicit ec: ExecutionContext)
+    extends Logging {
+
+  def submitBalanceRequest()(implicit hc: HeaderCarrier, request: DataRequest[_]): Future[Result] =
```

Comment:
It would help the reusablility of this method to return objects rather than a result e.g. Rather than redirect to the rate limit page we return a 'RateLimit' and let the controller handle the redirect with a matcher